### ISSUE TITLE
event_source_mapping support for batch windows of up to 5 minutes for functions with Amazon SQS

### ISSUE THREAD

**Issue by with-joy** [hashicorp/terraform-provider-aws#16503]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Lambda now supports batch windows of up to 5 minutes for functions with Amazon SQS as an event source. However, adding `maximum_batching_window_in_seconds` like below it throws an error:

```hcl
resource "aws_lambda_event_source_mapping" "event_source_mapping" {
  batch_size = 100

  event_source_arn = aws_sqs_queue.some_queue.arn
  enabled          = true
  function_name    = aws_lambda_function.example_lambda.arn
  
  maximum_batching_window_in_seconds = 1
}

```

```bash
Error: Error updating Lambda event source mapping: InvalidParameterValueException: Maximum batch window in seconds must be greater than 0 if maximum batch size is greater than 10
{
  {
     StatusCode: 400,
     RequestID: "8e591cea-c8b2-4077-9f7a-55675c59f0aa"
  },
  Message_: "Maximum batch window in seconds must be greater than 0 if maximum batch size is greater than 10",
  Type: "User"
}
```
Seems like the param is not passed in the payload during update, eg:
```bash
{"BatchSize":100,"Enabled":true,"FunctionName":"arn:aws:lambda:us-east-1:xxxxxxxx:function:example_lambda"}
```

**Please note** that there is no issue in creating the event_source_mapping, but only when updating it.

CLI works fine though:
```bash
aws lambda update-event-source-mapping \
    --uuid xxxx-xxxxx-xxxx-xxxxx-xxxxxxx \
    --enabled \
    --batch-size 100 \
    --maximum-batching-window-in-seconds 1

```

### New or Affected Resource(s)

* [aws_lambda_event_source_mapping](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_event_source_mapping)

### Potential Terraform Configuration

```hcl
resource "aws_lambda_event_source_mapping" "event_source_mapping" {
  batch_size = 100

  event_source_arn = aws_sqs_queue.some_queue.arn
  enabled          = true
  function_name    = aws_lambda_function.example_lambda.arn
  
  maximum_batching_window_in_seconds = 1
}

```

### References

* [Announcement](https://aws.amazon.com/about-aws/whats-new/2020/11/aws-lambda-now-supports-batch-windows-of-up-to-5-minutes-for-functions/)

**Comment by anGie44** [comment]
Hi @with-joy, thank you for raising this issue. Linking here the line in question that will need to be updated to account for sqs event sources on `Update` operations: 
https://github.com/hashicorp/terraform-provider-aws/blob/840888e8327f82f8abab32d536d2e7e13d37defa/aws/resource_aws_lambda_event_source_mapping.go#L367

**Comment by with-joy** [comment]
@anGie44 : thanks for the input. yes, I will try to open a PR with the change

**Comment by with-joy** [comment]
Hi @anGie44, looks like the other params (in the if block) are not supported by [SQS](https://docs.aws.amazon.com/cli/latest/reference/lambda/update-event-source-mapping.html). Therefore have moved out `MaximumBatchingWindowInSeconds ` outside of the `if-condition`.

**Comment by anGie44** [comment]
Thanks again @with-joy! This feature has been merged and will release with `v3.20.0` of the Terraform AWS provider.
